tests/roots/test-ext-autosummary/autosummary_dummy_complex_inheritance_module.py

```diff
@@ -0,0 +1,112 @@
+import sys
+
+class Parent:
+
+    relation = "Family"
+
+    def __init__(self):
+        self.name = "Andrew"
+        self.age = 35
+
+    def get_name(self):
+        return self.name
+
+    def get_age(self):
+        return f"Parent class -  {self.age}"
+
+    # Test mangled name behaviour
+    __get_age = get_age  # private copy of original get_age method
+    __private_parent_attribute = "Private_parent"
+
+
+class Child(Parent):
+
+    def __init__(self):
+        self.name = "Bobby"
+        self.age = 15
+
+    def get_name(self):
+        return self.name
+
+    def get_age(self):
+        return f"Child class -  {self.age}"
+
+    @staticmethod
+    def addition(a, b):
+        return a + b
+
+
+class Baby(Child):
+    
+    # Test a private attribute
+    __private_baby_name = "Private1234"
+
+    def __init__(self):
+        self.name = "Charlie"
+        self.age = 2
+
+    def get_age(self):
+        return f"Baby class - {self.age}"
+
+    class BabyInnerClass():
+        baby_inner_attribute = "An attribute of an inner class"
+
+
+class Job:
+
+    def __init__(self):
+        self.salary = 10
+
+    def get_salary(self):
+        return self.salary
+
+
+class Architect(Job):
+
+    def __init__(self):
+        self.salary = 20
+
+    def get_age(self):
+        return f"Architect age - {self.age}"
+
+# Test a class that inherits multiple classes
+class Jerry(Architect, Child):
+
+    def __init__(self):
+        self.name = "Jerry"
+        self.age = 25
+
+
+__all__ = ["Parent", "Child", "Baby", "Job", "Architect", "Jerry"]
+
+# The following is used to process the expected builtin members for different
+# versions of Python. The base list is for v3.11 (the latest testing when added)
+# and new builtin attributes/methods in later versions can be appended below.
```

Comment:
Can we have those values declared in the test instead? the test roots should be only contains the code we "test" (so this file should be like a real one on which you run autodoc).